lib/ansible/modules/cloud/amazon/iam_group.py

```diff
@@ -243,34 +246,37 @@ def create_or_update_group(connection, module):
     # Manage managed policies
     current_attached_policies = get_attached_policy_list(connection, module, params['GroupName'])
     if not compare_attached_group_policies(current_attached_policies, managed_policies):
+        current_attached_policies_arn_list = []
+        for policy in current_attached_policies:
+            current_attached_policies_arn_list.append(policy['PolicyArn'])
+
         # If managed_policies has a single empty element we want to remove all attached policies
         if purge_policy:
             # Detach policies not present
-            current_attached_policies_arn_list = []
```

Comment:
I'm not wholly convinced that the restructuring here is necessary for check mode - the wider changes here make this more difficult to review. Is there a reason that this change is more complicated than putting all the AWS write calls inside an `if module.check_mode` check?